src/roadmap/v1/lifecycle/app_streams.py

```diff
@@ -357,54 +278,118 @@ def app_streams_from_modules(
-            app_key = AppStreamKey(
-                name=app_stream_package.application_stream_name,
-                display_name=display_name,
-                application_stream_name=app_stream_package.application_stream_name,
-                start_date=app_stream_package.start_date,
-                end_date=app_stream_package.end_date,
-                os_major=os_major,
-                os_minor=os_minor if app_stream_package.rolling else None,
-                os_lifecycle=os_lifecycle if app_stream_package.rolling else None,
-                rolling=app_stream_package.rolling,
-                impl=AppStreamImplementation.package,
+        try:
+            relevant_app_streams.append(
+                RelevantAppStream(
+                    name=app_stream.name,
+                    display_name=app_stream.app_stream_entity.display_name,
+                    application_stream_name=app_stream.app_stream_entity.application_stream_name,
+                    start_date=app_stream.app_stream_entity.start_date,
+                    end_date=app_stream.app_stream_entity.end_date,
+                    os_major=app_stream.app_stream_entity.os_major,
+                    os_minor=app_stream.app_stream_entity.os_minor,
+                    impl=app_stream.app_stream_entity.impl,
+                    count=len(systems),
+                    rolling=app_stream.app_stream_entity.rolling,
+                    systems=systems,
+                    related=False,
+                )
             )
-            app_streams.add(app_key)
+        except Exception as exc:
+            raise HTTPException(detail=str(exc), status_code=400)
 
-    return app_streams
+    if related:
+        for app_stream in related_app_streams(systems_by_stream.keys()):
+            # Omit rolling app streams.
+            if app_stream.app_stream_entity.rolling:
+                continue
+
+            try:
+                relevant_app_streams.append(
+                    RelevantAppStream(
+                        name=app_stream.name,
+                        display_name=app_stream.app_stream_entity.display_name,
+                        application_stream_name=app_stream.app_stream_entity.application_stream_name,
+                        start_date=app_stream.app_stream_entity.start_date,
+                        end_date=app_stream.app_stream_entity.end_date,
+                        os_major=app_stream.app_stream_entity.os_major,
+                        os_minor=app_stream.app_stream_entity.os_minor,
+                        impl=app_stream.app_stream_entity.impl,
+                        count=0,
+                        rolling=app_stream.app_stream_entity.rolling,
+                        systems=[],
+                        related=True,
+                    )
+                )
+            except Exception as exc:
+                raise HTTPException(detail=str(exc), status_code=400)
+
+    if len(relevant_app_streams) == 0:
```

Comment:
There may be a reason for the explicit equality check. If not, it's more idiomatic to use a falsy check.
```suggestion
    if not relevant_app_streams:
```

Reply:
Is this the section that is adding some results if no related items were found? Probably worth putting a comment to explain that.

This should probably be handled in a separate PR since the behavior could be seen as surprising. It's worthy of a design discussion.

Reply:
this is on another branch now.

Reply:
fixed on other branch